### PR TITLE
Mt ad user auth part 2

### PR DESCRIPTION
**Pull request by annadolan**
@AliSchlereth @epintozzi 

Finished User Authentication stories, please take a look and merge if looks good!

@case-eee 

User story asks for navbar to say "Logged in as SOME_USER", but due to space issues, we have chosen to display the current user's name with "logout" (i.e. "Logout Casey"). Is this ok?

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<%= form_for @user do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name %>
+
+  <%= f.label :email %>
+  <%= f.text_field :email %>
+
+  <%= f.label :password %>
+  <%= f.text_field :password %>
+
```
Comment (epintozzi) [inline app/views/users/new.html.erb:12]: @annadolan @meganft  We should add a password confirmation field in this form with this next PR.

File: spec/features/sessions/user_can_log_in_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe "a user can log in" do
+  scenario "a user visits login page and logs in" do
+    user = User.create(name: "Bob", email: "cats@cats.cats", password: "cats")
+    visit login_path
+
+    fill_in "email", with: "#{user.email}"
+    fill_in "password", with: "#{user.password}"
```
Comment (epintozzi) [inline spec/features/sessions/user_can_log_in_spec.rb]: We should update this test after we add the password confirmation field and add a sad path test for if the passwords don't match or one is left blank.

File: spec/features/users/visitor_can_create_account_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    expect(current_path).to eq(new_user_path)
+  end
+
+  scenario "a visitor can create an account" do
+    visit new_user_path
+
+    fill_in "user[name]", with: "Bob"
+    fill_in "user[email]", with: "cats@cats.com"
+    fill_in "user[password]", with: "cats"
```
Comment (epintozzi) [inline spec/features/users/visitor_can_create_account_spec.rb:17]: We should update this test after we add the password confirmation field and add a sad path test for if the passwords don't match or one is left blank.

Comment (meganft) [reply spec/features/users/visitor_can_create_account_spec.rb:17]: @epintozzi  Agreed! I added waffle cards to add this sad path test and to add the password confirmation field.

**Pull request merged**